### ISSUE TITLE
createContentLoader do not return correct url of post

### ISSUE THREAD

**Issue by HelloYu** [vuejs/vitepress#3063]
### Describe the bug

I want to use `VitePress` to build a blog, I want to find current post base on route, so i have below code:
```
	function findCurrentIndex() {
		
		const result = posts.findIndex((p) => p.url.includes(route.path))
		if (result === -1) console.error(`blog post missing: ${route.path}`)
		return result
	}

```
this works fine, when we do not enable rewrites or set `base` configuration, because url of post comes from `createContentLoader` do not add `base` or prefix, so I wrap the url with `withBase` in many places, could anyone explain why `createContentLoader ` not return the 'correct' url base on site configuration or rewrites rules? 
I add `withBase` inside loader as well, but It doesn't work
```
export default createContentLoader(pattern, {
	excerpt: '<!--more-->',
	transform(raw): Post[] {
		return raw
			.map(({ url, frontmatter, excerpt }) => ({
				title: frontmatter.title,
				categories: frontmatter?.categories ?? [],
				description: excerpt,
				tags: frontmatter.tags ?? [],
				url: withBase(url), // in here will got error message: does not provide an export named 'withBase'
				excerpt,
				date: formatDate(frontmatter.date),
			}))
			.sort((a, b) => b.date.time - a.date.time)
	},
})
```

### Reproduction

as describe

### Expected behavior

createContentLoader return correct url

### System Info

```Text
System:
    OS: Windows 10 10.0.22621
    CPU: (4) x64 Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz
    Memory: 19.90 GB / 31.96 GB
  Binaries:
    Node: 18.17.1 - C:\Program Files\nodejs\node.EXE
    npm: 9.6.7 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.7.4 - ~\AppData\Roaming\npm\pnpm.CMD
  Browsers:
    Edge: Chromium (117.0.2045.60)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    vitepress: 1.0.0-rc.20 => 1.0.0-rc.20
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by HelloYu** [comment]
hi @brc-dd, can you help to provide some solutions?

**Comment by zonemeen** [comment]
Can you share your project repository? Or reproduce it on [StackBlitz](https://vitepress.new/).

**Comment by HelloYu** [comment]
> This was first fixed in [#2714](https://github.com/vuejs/vitepress/pull/2714), but then it was reverted since caused other problem, and now you have to convert the url manually.

Thanks
